### PR TITLE
Display contract features and docs links

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Display contract features and docs links.

## QA

- Visit [/advantage?test_backend=true](https://ubuntu-com-10357.demos.haus/advantage?test_backend=true) and log in.
- Click on the 'Features' and 'Documentation' tabs, you should see appropriate features/docs.
- Click on some different subscriptions on the left and check that they too have appropriate features/docs.

## Issue / Card

Fixes https://github.com/canonical-web-and-design/commercial-squad/issues/134.

## Screenshots

Screen Shot 2021-09-14 at 12 17 28 pm

Screen Shot 2021-09-14 at 12 18 56 pm

### CODE REVIEW

**Comment by huwshimi** [comment]
@wgx hopefully I've got this right. Each subscription is returned from the API with list of features and and each feature has a flag `enabled_by_default`. The features tab puts the `enabled_by_default: true` into the 'Included' list and the `enabled_by_default: false` into the 'Not included' list. The docs tab shows docs for the `enabled_by_default: true` features. Does that sound about right?

**Comment by wgx** [comment]
Hi @huwshimi it looks really good. 

There is another item this cycle called "Editable default enablements" which will change that "Features" tab to be editable - I am working on the design this week. The [basic notes live here](https://github.com/canonical-web-and-design/commercial-squad/issues/243#issuecomment-918112063) please take a look if you get 5 minutes. 

Quick note that free personal token still has renewal settings when it shouldn't have.

**Comment by wgx** [comment]
@huwshimi Were all the links @lechsandecki added to the Google Doc useful? There only appear to be a couple of docs links in the UI?

**Comment by ClementChaumel** [comment]
Small unrelated issue: when you select different contracts, the height of the app jumps around to accommodate the content but sometimes it messes up with the scroll-into-view thing

scrooll

File: static/js/src/advantage/react/components/Subscriptions/SubscriptionDetails/DetailsTabs/DetailsTabs.tsx [hunk 1]
```diff
@@ -38,30 +53,93 @@ const generateList = (title: string, items: ListItem[]) => (
...
+        case EntitlementType.CcEal:
+          link = {
+            label: "CC-EAL2 setup instructions",
+            url: "https://ubuntu.com/security/certifications/docs/cc",
+          };
+          break;
+        case EntitlementType.Fips:
+          link = {
+            label: "FIPS setup instructions",
+            url: "https://ubuntu.com/security/certifications/docs/fips ",
```
Comment (ClementChaumel) [inline static/js/src/advantage/react/components/Subscriptions/SubscriptionDetails/DetailsTabs/DetailsTabs.tsx:96]: Could these links be relative instead?

**Review by ClementChaumel — APPROVED** [review]
LGTM

**Comment by huwshimi** [comment]
> @huwshimi Were all the links @lechsandecki added to the Google Doc useful? There only appear to be a couple of docs links in the UI?

Yes they were great, thanks. The reason you're only seeing some links is that the API returns a list of the features for that subscription. In your "editable default enablements" issue you linked to those features equate to the list under "What comes with each plan:". At the moment I'm only showing the docs links for "default on" but I guess I should change that to listing the docs for features even if they are "default off"?

File: static/js/src/advantage/react/components/Subscriptions/SubscriptionDetails/DetailsTabs/DetailsTabs.tsx (hunk 1 above)
Comment (huwshimi) [reply static/js/src/advantage/react/components/Subscriptions/SubscriptionDetails/DetailsTabs/DetailsTabs.tsx:96]: Oh yeah, thanks!

**Comment by huwshimi** [comment]
> Quick note that free personal token still has renewal settings when it shouldn't have.

Thanks



> > @huwshimi Were all the links @lechsandecki added to the Google Doc useful? There only appear to be a couple of docs links in the UI?
> 
> @wgx Yes they were great, thanks. The reason you're only seeing some links is that the API returns a list of the features for that subscription. In your "editable default enablements" issue you linked to those features equate to the list under "What comes with each plan:". At the moment I'm only showing the docs links for "default on" but I guess I should change that to listing the docs for features even if they are "default off"?

@wgx I've updated the PR to show both default on and off and also hidden all docs links from the free plan. Does that look right now?

**Comment by wgx** [comment]
@huwshimi this looks really great now, thanks!

**Pull request merged**